SpatialGDK/Source/SpatialGDK/Private/SimulatedPlayers/SimulatedPlayer.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SimulatedPlayer.h"
+
+#include "Engine.h"
+#include "Misc/Parse.h"
+#include "Misc/CommandLine.h"
+#include "SpatialConstants.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogSimulatedPlayer, Log, All);
+
+bool USimulatedPlayer::IsSimulatedPlayer(UObject* WorldContextObject)
+{
+	UEngine* engine = GEngine;
+	if (engine == nullptr)
+	{
+		UE_LOG(LogSimulatedPlayer, Error, TEXT("Cannot check if player is simulated: GEngine is null. Defaulting to false."));
+		return false;
+	}
+
+	UWorld* world = engine->GetWorldFromContextObject(WorldContextObject, EGetWorldErrorMode::ReturnNull);
```

Comment:
Error handling in this function is too heavy handed. Refactor please.

`GEngine` better be non-null otherwise we have bigger problems. So just do what `UGameplayStatics::GetGameInstance()` does, and in failure print a `Warning` not `Error`.